src/meshapi/util/task_utils.py

```diff
@@ -0,0 +1,21 @@
+import logging
+
+import boto3
+
+
+# Returns the key to the most recently modified object in an S3 bucket
+def get_most_recent_object(bucket_name: str, prefix: str) -> str | None:
+    s3_client = boto3.client("s3")
+    objects = s3_client.list_objects_v2(Bucket=bucket_name, Prefix=prefix)
+
+    if "Contents" not in objects:
+        logging.error(f"No objects found in '{bucket_name}/{prefix}'.")
+        return None
+
+    most_recent_object = max(objects["Contents"], key=lambda obj: obj["LastModified"])
```

Comment:
Hmm this isn't going to work, S3 is paginated. You have to loop through an iterator given to you by `boto3` to hit all the objects if you want to do this, but that's pretty inefficient. A more efficient solution would be to include the date in the backup prefix, like: `s3://meshdb-data-backups/meshdb-backups/prod1/2024/09/06/default-meshdb-celery-59496585dd-75vqq-2024-08-30-230003.psql.bin`. Then you can just list todays backups using the prefix and guarantee it will fit on one page

Reply:
Shit. I knew there was gonna be some issue like that. Unfortunately I have zero control over precisely how dbbackup stores backups in S3. We're still on my fork of it, so maybe I could patch it. Is there a way to ask boto for a specific time range or something?

Reply:
Nope, that's a limitation of S3, it's a flat store where all the info you care about needs to be encoded into the object key

Reply:
I suppose one thing you have going for you is that I set the bucket to delete backups older than 90 days, so this will be a constant number of objects rather than continuously increasing

Reply:
I think it actually might work if you set `prefix` to `s3://meshdb-data-backups/meshdb-backups/prod1/default-meshdb-celery-59496585dd-75vqq-YYYY-MM-DD`. Files are not special in S3, they're just a part of the key

Reply:
Problem with that is `meshdb-celery-59496585dd-75vqq` is a pod name, which is non-deterministic, ephemeral, and (almost) impossible to get (especially when the most recent backup may have happened 59 minutes ago). If I could somehow specify a wildcard with a date, that could work.

Reply:
No wildcard support :( just prefixes

Reply:
Ah, wait a sec: https://django-dbbackup.readthedocs.io/en/stable/configuration.html#dbbackup-filename-template

Reply:
The forward slashes would make the console UI nicer, but that's the only difference. Doesn't impact performance or anything

Reply:
oh, also, if you don't specify a backup file, the latest one gets used: https://github.com/jazzband/django-dbbackup/blob/master/dbbackup/management/commands/_base.py#L111

¯\\\_(ツ)_/¯